src/bthread/task_tracer.cpp

```diff
@@ -356,15 +356,18 @@ void TaskTracer::SignalHandler(int, siginfo_t* info, void* context) {
     // Ref has been taken before the signal is sent, so no need to add ref here.
     butil::intrusive_ptr<SignalSync> signal_sync(
         static_cast<SignalSync*>(info->si_value.sival_ptr), false);
-    if (NULL == signal_sync || NULL == signal_sync->result) {
+    if (NULL == signal_sync) {
         // The signal is not from Tracer, such as TaskControl, do nothing.
         return;
     }
-    Result* result = signal_sync->result;
+    void *pcs[Result::MAX_TRACE_NUM];
     // Skip the first frame, which is the signal handler itself.
-    result->frame_count = absl::DefaultStackUnwinder(&result->ips[0], NULL,
-                                                     arraysize(result->ips),
-                                                     1, context, NULL);
+    int depth = absl::DefaultStackUnwinder(pcs, NULL, arraysize(pcs), 1,
+                                           context, NULL);
+    signal_sync->pcs.reserve(depth);
```

Comment:
Allocating memory in a signal handler is dangerous. Maybe we should make the type of result in SignalSync Result instead of Result*, just like before.